scripts/variantstore/wdl/GvsCreateVAT.wdl

```diff
@@ -604,3 +619,72 @@ task BigQueryExportVat {
     }
 }
 
+task MergeVatTSVs {
+    input {
+        Array[Boolean] export_done
+        Array[String] contig_array
+        String project_id
+        String output_path
+
+        Int? merge_vcfs_disk_size_override
+        String? service_account_json_path
+    }
+
+    Int disk_size = if (defined(merge_vcfs_disk_size_override)) then merge_vcfs_disk_size_override else 50
+    String has_service_account_file = if (defined(service_account_json_path)) then 'true' else 'false'
+
+    command <<<
+        apt-get update
+        apt-get install tabix
+
+        # custom function to prepend the current datetime to an echo statement "borrowed" from ExtractAnAcAfFromVCF
+        echo_date () { echo "`date "+%Y/%m/%d %H:%M:%S"` $1"; }
+
+        if [ ~{has_service_account_file} = 'true' ]; then
+            gsutil cp ~{service_account_json_path} local.service_account.json
+            gcloud auth activate-service-account --key-file=local.service_account.json
+            gcloud config set project ~{project_id}
+        fi
+
+        mkdir TSVs
+        contigs=( ~{sep=' ' contig_array} )
+        files=" header.gz"
+
+        echo_date "looping over contgs: $contigs"
+        for i in "${contigs[@]}"
+        do
+            echo_date "copying files from ~{output_path}export/$i/*.tsv.gz"
+            gsutil -m cp ~{output_path}export/$i/*.tsv.gz TSVs/
+            echo_date "concatenating local tsv.gz files"
+            cat TSVs/*.tsv.gz > vat_$i.tsv.gz
```

Comment:
does ordering not matter within the contigs?

Reply:
there's an `ORDER BY position` in the query that generates the TSV files within the contig folders

Reply:
Let me explain what I'm thinking of more... lets say that the export generates 3 TSVs for a particular contig: `001.tsv.gz`, `002.tsv.gz` and `003.tsv.gz`.  The ORDER BY will make sure that each file is sorted locally (ie within the file) and that the shards (the 001, 002, 003) are also ordered (ie 001 is before 002).

But when you `cat *.tsv.gz` your relying on the glob expansion in the shell to turn that into `cat file1 file2 file3` and the order of those files will be the order in which they are concatenated.  It's non-deterministic.

I think you have to get that ordered list of files and then cat them explicitly in that order, either with `cat file1 >> out`, `cat file2 >> out` and so on or even just the explicit `cat file1 file2 file3` 

Happy to slack-chat if that still doesn't make sense (or it's easier to tell me why it's not a problem) there!

Reply:
Ah, ok — I didn't realize that `cat` was  non-deterministic.

Reply:
yeah -- and specifically it's not `cat` it's the `*` 

Although TIL about this, so maybe it's ok?  https://serverfault.com/questions/122737/in-bash-are-wildcard-expansions-guaranteed-to-be-in-order

Reply:
going to keep it using `*` and reference that article in a comment for future concerns